src/datastore/fetchingFunctions.ts

```diff
@@ -23,12 +23,17 @@ import {
 export const fetchRiddleComplexDetail = async (
 	linkId: string,
 	user: User
-): Promise<RiddleDisplayDetail> => {
+): Promise<RiddleDisplayDetail | undefined> => {
 	const riddleInfo = (
 		await fetchUserRiddleInfo(linkId, user?.email ?? '')
 	)?.data();
 	const riddleRes = await fetchRiddle(linkId);
```

Comment:
Tady si můžeme ušetřit práci a přehodit pořadí fetchování. Pokud je `riddleRes ` undefined, můžeme vrátit a až pak fetchnout `riddleInfo = (await fetchUserRiddleInfo(...))`, protože o neexistující hádance určitě nemáme záznamy řešení.

Reply:
To je výborný point!